### PR TITLE
Remove members from subgroup

### LINKED ISSUE #8: Remove members of subgroup
As a member in a group chat
I want the ability to remove members of a subgroup
So that I can modify the list as time goes

```
# Command
/kick [name of the exiting group] [@Carl @Doug]

# Response
@Carl and @Doug was removed from [name of existing group]
```

### PR DESCRIPTION
**Pull request by AshciR**
closes #8

### CODE REVIEW

File: src/shout_subgroup/modify_subgroup.py [hunk 1]
```diff
@@ -187,3 +186,99 @@ async def subgroup_handler(update: Update, context: ContextTypes.DEFAULT_TYPE) -
     except Exception:
         logging.exception("An unexpected exception occurred")
         await update.message.reply_text("Whoops 😅, something went wrong on our side.")
+
+
+async def remove_users_from_existing_subgroup(
```
Comment (AshciR) [inline src/shout_subgroup/modify_subgroup.py:191]: Focus

File: src/shout_subgroup/modify_subgroup.py [hunk 2]
```diff
@@ -187,3 +186,99 @@ async def subgroup_handler(update: Update, context: ContextTypes.DEFAULT_TYPE) -
...
+        return
+
+    subgroup_name = args[0]
+
+    usernames = {name.replace("@", "") for name in set(args[1:])}  # removing mention from args for usernames
+    if not await usernames_valid(usernames):
+        await update.message.reply_text("Not all the usernames are valid. Please re-check what you entered.")
+    try:
+
+        subgroup = await remove_users_from_existing_subgroup(session, update.effective_chat.id, subgroup_name, usernames)
```
Comment (AshciR) [inline src/shout_subgroup/modify_subgroup.py:256]: I didn't pull this into a `_handle_remove_subgroup_member` b/c I don't think the number of lines of logic needed to be abstracted

File: tests/test_remove_subgroup_members.py
Comment (AshciR) [inline tests/test_remove_subgroup_members.py:1]: Added 7 test cases

File: src/shout_subgroup/modify_subgroup.py [hunk 3]
```diff
@@ -187,3 +186,99 @@ async def subgroup_handler(update: Update, context: ContextTypes.DEFAULT_TYPE) -
...
+        msg = f"Subgroup {subgroup_name} should exist for telegram group chat id {telegram_chat_id}."
+        logging.info(msg)
+        raise SubGroupDoesNotExistsError(msg)
+
+    # If we can't find all the users, then it means we have not saved them yet.
+    users_to_be_removed = await find_users_by_usernames(db, usernames)
+    if len(users_to_be_removed) != len(usernames):
+        raise UserDoesNotExistsError("All the usernames are not in the database.")
+
+    # Find all the users who aren't in the group, then add them
```
Comment (rick1e) [inline src/shout_subgroup/modify_subgroup.py:213]: i think this comment wrong.
shouldn't it be to find the users in the group and remove them

**Review by rick1e — APPROVED** [review]

**Pull request merged**